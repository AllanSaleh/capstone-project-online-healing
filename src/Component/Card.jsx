import React from 'react';

const Card = ({ title, content, showIcon, icon, buttonText, showButton }) => {
  return (
    <div className="flex flex-col max-w-sm items-center shadow-lg py-8 px-1">
      {showIcon && <img src={icon} alt="" className="flex-initial pb-4" />}
      <h3 className="text-center text-title pb-1">{title}</h3>
      <p className="text-gray-500 text-paragraph pb-4">{content}</p>
      {showButton && (
        <button className="bg-blue-dark py-2 px-4 text-btn text-black hover:bg-opacity-50 transition duration-300 rounded-md">
          {buttonText}
        </button>
      )}
    </div>
  );
};

export default Card;
