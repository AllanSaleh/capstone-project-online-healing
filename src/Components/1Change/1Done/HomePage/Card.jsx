import React from 'react';

const Card = ({ title, content, showIcon, icon, buttonText, showButton }) => (
  <div className="flex flex-col max-w-sm items-center shadow-lg py-8 px-16">
    {showIcon && <img src={icon} alt="" className="flex-initial pb-4" />}
    <h3 className="text-center text-title pb-1 capitalize">{title}</h3>
    <p className="text-opacity-50 text-black text-center text-paragraph pb-4">{content}</p>
    {showButton && (
      <button
        type="button"
        className="bg-blue-dark py-2 px-4 text-btn text-black hover:bg-opacity-50 transition duration-300 rounded-md"
      >
        {buttonText}
      </button>
    )}
  </div>
);

export default Card;
