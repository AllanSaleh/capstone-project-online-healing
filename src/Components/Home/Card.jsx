import React from 'react';

// eslint-disable-next-line react/prop-types
const Card = ({ title, content, showIcon, icon, buttonText, showButton }) => (
  <div className="flex flex-col max-w-sm items-center shadow-lg rounded-lg h-full w-11/12 md:w-1/3 my-6 md:my-0 md:mx-2 p-2 lg:py-6 lg:px-12">
    {showIcon && <img src={icon} alt="" className="flex-initial mb-4" />}
    <h3 className="text-center text-xl  lg:text-3xl 2xl:text-title pb-1 capitalize">
      {title}
    </h3>
    <p className="text-opacity-50 text-black text-center text-sm md:text-md lg:text-paragraph py-4">
      {content}
    </p>
    {showButton && (
      <button
        type="button"
        className="bg-blue-dark py-2 px-4 uppercase text-paragraph border text-black hover:bg-transparent hover:border-blue-dark hover:text-blue-dark  transition-all duration-200 rounded-md"
      >
        {buttonText}
      </button>
    )}
  </div>
);

export default Card;
