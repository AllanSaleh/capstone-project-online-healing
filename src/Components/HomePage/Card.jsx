import React from 'react';
import { useHistory } from 'react-router-dom';

export default function Card({ title, content, showIcon, icon, buttonText, showButton }) {
  const history = useHistory();

  const GoToRoute = () =>
    history.push({
      pathname: '/Purchase',
      state: { ticket: title, price: content },
    });

  return (
    <div className="h-full w-full my-4 flex flex-col items-center justify-evenly shadow-lg rounded-lg md:w-1/4 p-4">
      {showIcon && <img src={icon} alt="" className="flex-initial mb-4" />}
      <h3 className="text-center text-xl  lg:text-3xl 2xl:text-title pb-1 capitalize">
        {title} Tickets
      </h3>
      <p className="text-opacity-50 text-black text-center text-sm md:text-md lg:text-paragraph py-4">
        {content}$
      </p>
      {showButton && (
        <button
          onClick={() => GoToRoute()}
          type="button"
          className="bg-blue-dark py-2 px-4 uppercase text-paragraph border text-black hover:bg-transparent hover:border-blue-dark hover:text-blue-dark  transition-all duration-200 rounded-md"
        >
          {buttonText}
        </button>
      )}
    </div>
  );
}
