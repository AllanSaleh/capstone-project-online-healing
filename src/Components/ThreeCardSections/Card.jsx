import React from 'react';

// eslint-disable-next-line react/prop-types
const Card = ({ title, content, icon, buttonText }) => (
  <div className="flex flex-col max-w-sm items-center shadow-lg py-8 px-16">
    {icon && <img src={icon} alt="" className="flex-initial pb-4" />}
    <h3 className="text-center text-title pb-1 capitalize">{title}</h3>
    {buttonText ? (
      <p className="text-opacity-50 text-black text-center text-4xl pb-4">{content}</p>
    ) : (
      <p className="text-opacity-50 text-black text-center text-paragraph pb-4">{content}</p>
    )}
    {buttonText && (
      // eslint-disable-next-line react/button-has-type
      <button className="bg-blue-dark py-2 px-4 text-subtitle border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition-all duration-300 rounded-lg uppercase">
        {buttonText}
      </button>
    )}
  </div>
);

export default Card;
