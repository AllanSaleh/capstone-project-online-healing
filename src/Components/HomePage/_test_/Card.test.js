import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Card from '../Card';

it(' Card renders correctly', () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});
