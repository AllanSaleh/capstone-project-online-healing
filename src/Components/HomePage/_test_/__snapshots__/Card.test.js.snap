// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Card renders correctly 1`] = `
<div
  className="h-full w-full my-4 flex flex-col items-center justify-evenly shadow-lg rounded-lg md:w-1/4 p-4"
>
  <h3
    className="text-center text-xl  lg:text-3xl 2xl:text-title pb-1 capitalize"
  >
     Tickets
  </h3>
  <p
    className="text-opacity-50 text-black text-center text-sm md:text-md lg:text-paragraph py-4"
  >
    $
  </p>
</div>
`;
